gcp-cloud-function/massdriver.yaml

```diff
@@ -25,7 +25,115 @@ params:
     - endpoint
   properties:
     cloud_function_configuration:
-      $ref: https://raw.githubusercontent.com/massdriver-cloud/artifact-definitions/main/definitions/types/gcp-cloud-function-configuration.json
+      type: object
```

Comment:
Exposing the runtimes here (and other configs) because they were hidden before, not sure why. This change allows the user to deploy a cloud function without their code files. It will just deploy a hello world with the desired runtime. Of course, user could update the underlying bundle to use whatever they want as their source for function code. 

Specifying the runtime is also oddly required for GCP cloud function:
![image](https://github.com/massdriver-cloud/application-templates/assets/8037512/1d82b741-a270-4811-a9d5-b9c01c7df8ed)

Open to suggestions if this isn't the best approach.